### PR TITLE
Fix #3099: add bootstrap table style and remove outdated syntax

### LINKED ISSUE #3099: Refine Forms Administration table
### Describe the bug

Forms Administration used legacy style and outdated syntax.

### To Reproduce

1. Choose Administrator > Forms 
2. Click Forms Administration
3. See the table

Untitled

### Client configuration

- Browser: Firefox 73.0.1（64 bit）
- Operating system: Windows 10

### PR DESCRIPTION
**Pull request by stu01509**
Fixes #3099

#### Short description of what this resolves:
Add bootstrap table style.
Remove outdated syntax.

image

image

### CODE REVIEW

**Comment by stu01509** [comment]
I just add the Bootstrap table style and remove outdated syntax.
I think this section still has room for improvement especially the td color.

File: interface/forms_admin/forms_admin.php [hunk 1]
```diff
@@ -46,195 +46,217 @@
...
+                                echo "<td>";
+                                echo "<select name='aco_spec_" . attr($registry['id']) . "'>";
+                                echo "<option value=''></option>";
+                                echo AclExtended::genAcoHtmlOptions($priority_category['aco_spec']);
+                                echo "</select>";
+                                echo "</td>";
+                                ?>
+                            </tr>
+                                <?php
+                                if ($tdStyle=="bg-dark text-light") {
```
Comment (im-Amitto) [inline interface/forms_admin/forms_admin.php]: $tdStyle = $tdStyle == "bg-dark text-light" ? "bg-light" : "bg-dark text-light";

Just a guy in love with ternary operator :)

File: interface/forms_admin/forms_admin.php [hunk 2]
```diff
@@ -46,195 +46,217 @@
...
+                                </td>
+                                <td class="<?php echo $tdStyle?>">
+                                    <span class=text><?php echo xlt($phpState); ?></span>
+                                </td>
+                                <td class="<?php echo $tdStyle?>">
+                                    <span class=text><?php echo xlt('n/a'); ?></span>
+                                </td>
+                            </tr>
+                            <?php
+                                if ($tdStyle=="bg-dark text-light") {
```
Comment (im-Amitto) [inline interface/forms_admin/forms_admin.php]: Your choice if you wanna use ternary here :)

File: interface/forms_admin/forms_admin.php (hunk 1 above)
Comment (stu01509) [reply interface/forms_admin/forms_admin.php]: The ternary operator is a good choice,
But I just follow the same coding style lol

File: interface/forms_admin/forms_admin.php [hunk 3]
```diff
@@ -46,195 +46,217 @@
...
-</table>
-<hr>
+            <?php //REGISTERED SECTION ?>
+            <span class="font-weight-bold"><?php echo xlt('Registered');?></span>
+            <form method="post" action ='./forms_admin.php'>
+                <span class="font-italic">
+                    <?php echo xlt('click here to update priority, category, nickname and access control settings'); ?>
+                </span>
+                <input type="hidden" name="csrf_token_form" value="<?php echo attr(CsrfUtils::collectCsrfToken()); ?>" />
+                <input class="btn btn-primary" type='submit' name='update' value='<?php echo xla('Update'); ?>'>
```
Comment (bradymiller) [inline interface/forms_admin/forms_admin.php]: It seems like this should be `Save` instead of `Update`, since it is saving stuff.

**Comment by bradymiller** [comment]
hi @stu01509 ,

The dark colors mixed with white colors is making it harder me to follow the rows. Would recommend dropping the dark background. This will make it easier for user to follow the flow to add a new form, which consist of hitting `Register` on a form at the bottom section, of which that form then goes to the top section. They then click `Install DB` which is a link that replaces `DB Installed` when database items for the form haven't been installed yet. They then click `disabled`, which then changes to `enabled`.

Something to think about would be changing the enabled/disabled field to a switch https://getbootstrap.com/docs/4.2/components/forms/#switches
(as an aside, @stu01509 , @im-Amitto , @tywrenn is this switch considered a standard now for checkboxes and such, or is it best to stay away from this?)

**Comment by stu01509** [comment]
Hi @bradymiller 

I got it, I will remove the dark color mixed style and move Unregistered form to the top.

As for the Bootstrap switch, I'm so like it, it suitable for our interface style, but I'm not sure it is the best choice, Maybe @im-Amitto  and @tywrenn have a different idea.

**Comment by im-Amitto** [comment]
> Something to think about would be changing the enabled/disabled field to a switch https://getbootstrap.com/docs/4.2/components/forms/#switches
> (as an aside, @stu01509 , @im-Amitto , @tywrenn is this switch considered a standard now for checkboxes and such, or is it best to stay away from this?)

Switch can be seen as checkbox with a better UI as it does use a custom checkbox within it but still i won't prefer to replace every checkbox with a toggle switch.
A toggle switch can be seen as a physical switch that allows users to turn things on or off, like a light switch.
A checkbox can easily represent three states: unselected, selected, and indeterminate. The last state represents a situation where a list of sub-options is grouped under a parent option and sub-options are in both selected and unselected states.
In our case using switch should't be a problem and it's currently being used in almost every mobile OS so we don't need to worry much about it.
P.S.: I really thought of keeping it short

**Comment by stu01509** [comment]
Hi @bradymiller

I removed the dark background-color. But I'm not sure 

Does it mean Unregistered form should move top of the registered form?

>  which consist of hitting Register on a form at the bottom section,
 of which that form then goes to the top section. 

Can you describe more about this detail?
> They then click Install DB which is a link that replaces DB Installed when database items for the form haven't been installed yet. They then click disabled, which then changes to enabled

image

image

**Comment by bradymiller** [comment]
hi @stu01509 ,
No need to move things around. Just pointing out the flow to ensure they were seen (in the dark background for example, some of those links were hidden). Screenshots look good. When I get to a computer, I'll check out the code and then likely bring it into codebase.

**Comment by stu01509** [comment]
Hi @bradymiller,

Okay, Waiting for your review.
If any requested change please let me know, Thanks.

**Comment by bradymiller** [comment]
hi @stu01509 , Code looks good and testing looks great, so bringing into codebase. Thanks for the improvement! -brady
(for future, see if you can get your editor to support psr2, which is why travis is failing; this is super easy for me to fix via a one-liner, which will do in subsequent commit)

**Comment by stu01509** [comment]
> hi @stu01509 , Code looks good and testing looks great, so bringing into codebase. Thanks for the improvement! -brady
> (for future, see if you can get your editor to support psr2, which is why travis is failing; this is super easy for me to fix via a one-liner, which will do in subsequent commit)

Hi @bradymiller, 

Thanks for your mentions, I already install psr2 plugin in my editor :)

File: interface/forms_admin/forms_admin.php [hunk 4]
```diff
@@ -46,195 +46,205 @@
...
+                                    <?php
+                                    if ($phpState == "PHP extracted") {
+                                        echo '<a class=link_submit href="./forms_admin.php?name=' . attr_url($fname) . '&method=register&csrf_token_form=' . attr_url(CsrfUtils::collectCsrfToken()) . '">' . xlt('register') . '</a>';
+                                    } else {
+                                        echo '<span class=text>' . xlt('n/a') . '</span>';
+                                    }
+                                    ?>
+                                </td>
+                                <td>
+                                    <span class=text><?php echo xlt($phpState); ?></span>
```
Comment (tywrenn) [inline interface/forms_admin/forms_admin.php]: Should be in quotes

File: interface/forms_admin/forms_admin.php [hunk 5]
```diff
@@ -46,195 +46,205 @@
...
+                                </td>
+                                <td>
+                                    <span class=text><?php echo xlt('n/a'); ?></span>
```
Comment (tywrenn) [inline interface/forms_admin/forms_admin.php]: Should be in quotes

File: interface/forms_admin/forms_admin.php [hunk 6]
```diff
@@ -46,195 +46,205 @@
...
+                                    "select priority, category, nickname, aco_spec from registry where id = ?",
+                                    array($registry['id'])
+                                );
+                                ?>
+                            <tr>
+                                <td>
+                                    <span class='text'><?php echo text($registry['id']); ?></span>
+                                </td>
+                                <td>
+                                    <span class='bold'><?php echo text(xl_form_title($registry['name'])); ?></span>
```
Comment (tywrenn) [inline interface/forms_admin/forms_admin.php]: Should be `font-weight-bold`

**Pull request merged**